rpc/v2/types.go

```diff
@@ -126,6 +128,7 @@ type CreateValidatorMsg struct {
 	SecurityContact    string                    `json:"securityContact"`
 	Details            string                    `json:"details"`
 	SlotPubKeys        []bls.SerializedPublicKey `json:"slotPubKeys"`
+	SlotKeySigs        []bls.SerializedSignature `json:"slotKeySigs"`
```

Comment:
you added new field here? Is it mandatory?  It will it break a lot of our API and applications like the hmycli in the https://github.com/harmony-one/go-sdk repo.  And our staking dashboard?

Reply:
It is mandatory. To create a new validator, a `slotKeySigs` must be provided:

https://github.com/renlulu/harmony/blob/feat/rosetta-staking/staking/types/messages.go#L56
https://github.com/renlulu/harmony/blob/feat/rosetta-staking/rosetta/services/construction_create.go#L127

```
// CreateValidator - type for creating a new validator
type CreateValidator struct {
	ValidatorAddress   common.Address `json:"validator-address"`
	Description        `json:"description"`
	CommissionRates    `json:"commission"`
	MinSelfDelegation  *big.Int                  `json:"min-self-delegation"`
	MaxTotalDelegation *big.Int                  `json:"max-total-delegation"`
	SlotPubKeys        []bls.SerializedPublicKey `json:"slot-pub-keys"`
	SlotKeySigs        []bls.SerializedSignature `json:"slot-key-sigs"`
	Amount             *big.Int                  `json:"amount"`
}
```

And for `hmycli`, it doesn't use `CreateValidatorMsg`, instead, it use `CreateValidator` directly:

https://github.com/harmony-one/go-sdk/blob/master/cmd/subcommands/staking.go#L398


```
	delegateStakePayloadMaker := func() (staking.Directive, interface{}) {
				return staking.DirectiveCreateValidator, staking.CreateValidator{
					address.Parse(validatorAddress.String()),
					desc,
					staking.CommissionRates{
						commisionRate,
						commisionMaxRate,
						commisionMaxChangeRate},
					minSelfDel.RoundInt(),
					maxTotalDel.RoundInt(),
					blsPubKeys,
					blsSigs,
					amt.RoundInt(),
				}
			}
```

About the dashboard, can you point me out which repo it is, I can check it out, thanks!

Reply:
Even for hmycli, it helps users sign on a predefine message: https://github.com/harmony-one/go-sdk/blob/master/pkg/keys/bls.go#L226 then pass it to `CreateValidator`.